yt_dlp/extractor/niconico.py

```diff
@@ -367,229 +388,120 @@ def parse_format_id(id_code):
-            elif error_code == 'NOT_FOUND':
-                raise ExtractorError('The video is not found.',
-                                     expected=True)
-            elif error_code == 'COMMUNITY':
-                self.to_screen('%s: The video is community members only.' % video_id)
-            else:
-                raise ExtractorError('%s reports error: %s' % (self.IE_NAME, error_code))
+        try:
+            webpage, handle = self._download_webpage_handle(
+                'http://www.nicovideo.jp/watch/' + video_id, video_id)
+            if video_id.startswith('so'):
+                video_id = self._match_id(handle.geturl())
+
+            api_data = self._parse_json(self._html_search_regex(
+                'data-api-data="([^"]+)"', webpage,
+                'API data', default='{}'), video_id)
+        except ExtractorError as e:
+            try:
+                api_data = self._download_json(
+                    'https://www.nicovideo.jp/api/watch/v3/%s?_frontendId=6&_frontendVersion=0&actionTrackId=AAAAAAAAAA_%d' % (video_id, time_millis()), video_id,
+                    note='Downloading API JSON', errnote='Unable to fetch data')['data']
+            except (ExtractorError, KeyError):
+                if not isinstance(e.cause, compat_HTTPError):
+                    raise e
+                else:
+                    e = e.cause
+                webpage = e.read().decode('utf-8', 'replace')
+                error_msg = self._html_search_regex(
+                    r'(?s)<section\s+class="(?:(?:ErrorMessage|WatchExceptionPage-message)\s*)+">(.+?)</section>',
+                    webpage, 'error reason', group=1, default=None)
+                if not error_msg:
+                    raise e
+                else:
+                    error_msg = re.sub(r'\s+', ' ', error_msg)
+                    raise ExtractorError(error_msg, expected=True)
 
-        # Start extracting video formats
         formats = []
 
-        # Get HTML5 videos info
-        quality_info = try_get(api_data, lambda x: x['media']['delivery']['movie'])
-        if not quality_info:
-            raise ExtractorError('The video can\'t be downloaded', expected=True)
-
-        for audio_quality in quality_info.get('audios') or {}:
-            for video_quality in quality_info.get('videos') or {}:
-                if not audio_quality.get('isAvailable') or not video_quality.get('isAvailable'):
-                    continue
-                formats.append(self._extract_format_for_quality(
-                    api_data, video_id, audio_quality, video_quality))
-
-        # Get flv/swf info
-        timestamp = None
-        video_real_url = try_get(api_data, lambda x: x['video']['smileInfo']['url'])
-        if video_real_url:
-            is_economy = video_real_url.endswith('low')
+        def get_video_info(items, get_first=True):
+            return traverse_obj(api_data, ('video', items), get_all=not get_first)
```

Comment:
```suggestion
        def get_video_info(*items, get_first=True, **kwargs):
            return traverse_obj(api_data, ('video', *items), get_all=not get_first, **kwargs)
```
would make it possible to handle `thumbnail`, `view_count`, `comment_count` too

Eg:
```diff
- thumbnail = traverse_obj(api_data, ('video', 'thumbnail', 'url'))
+ thumbnail = get_video_info(api_data, 'thumbnail', 'url')